idunn/datasources/hove/models.py

```diff
@@ -0,0 +1,338 @@
+"""
+Derialize Hove's responses and conversion primitives following the following equivalence:
+ - route <-> journey
+ - leg <-> section
+ - step <-> single element of section ? Can be re-cut later ?
+"""
+
+import logging
+from datetime import datetime
+from enum import Enum
+from typing import Iterator, List, Optional, Tuple
+
+from geojson_pydantic.geometries import Geometry, LineString
+from geojson_pydantic.types import Position
+from pydantic.class_validators import validator
+from pydantic.fields import Field
+from pydantic import BaseModel
+
+import idunn.directions.models as api
+from idunn.directions.models import IdunnTransportMode
+
+logger = logging.getLogger(__name__)
+
+
+class Distances(BaseModel):
+    bike: int
+    car: int
+    ridesharing: int
+    taxi: int
+    walking: int
+
+    def overall(self):
+        return self.bike + self.car + self.ridesharing + self.taxi + self.walking
+
+
+class Coordinate(BaseModel):
+    lat: float
+    lon: float
+
+    def to_position(self) -> Position:
+        return (self.lon, self.lat)
+
+
+class Instruction(BaseModel):
+    direction: int
+    duration: int
+    instruction: Optional[str]
+    instruction_start_coordinate: Optional[Coordinate]
+    length: int
+    name: str
+
+    def get_api_modifier(self) -> str:
+        # See https://docs.api.com/api/navigation/directions/#step-maneuver-object
```

Comment:
Should `https://docs.api.com` be replaced by the actual hove documentation URL ?

Reply:
:laughing:

Reply:
It to me a minute to understand what happened here, the actual link is https://docs.mapbox.com/api/navigation/directions/#step-maneuver-object

> I'll search and replace "mapbox" -> "api", what could go wrong here?

I'll fix this and add some doc to explain the role of this function.